Kitodo/src/main/java/org/kitodo/production/interfaces/activemq/KitodoScriptProcessor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.interfaces.activemq;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import javax.jms.JMSException;
+
+import org.kitodo.config.ConfigCore;
+import org.kitodo.config.enums.ParameterCore;
+import org.kitodo.data.database.beans.Process;
+import org.kitodo.data.database.exceptions.DAOException;
+import org.kitodo.data.exceptions.DataException;
+import org.kitodo.exceptions.InvalidImagesException;
+import org.kitodo.exceptions.MediaNotFoundException;
+import org.kitodo.exceptions.ProcessorException;
+import org.kitodo.production.services.ServiceManager;
+import org.kitodo.production.services.command.KitodoScriptService;
+import org.kitodo.production.services.data.ProcessService;
+
+/**
+ * Executes instructions to start a Kitodo Script command from the Active MQ
+ * interface. The MapMessage must contain the command statement in the
+ * {@code script} argument. You pass a list of the process IDs as
+ * {@code processes}.
+ */
+public class KitodoScriptProcessor extends ActiveMQProcessor {
+
+    private final KitodoScriptService kitodoScriptService = ServiceManager.getKitodoScriptService();
+    private final ProcessService processService = ServiceManager.getProcessService();
+
+    public KitodoScriptProcessor() {
+        super(ConfigCore.getOptionalString(ParameterCore.ACTIVE_MQ_KITODO_SCRIPT_QUEUE).orElse(null));
+    }
+
+    @Override
+    protected void process(MapMessageObjectReader ticket) throws ProcessorException, JMSException {
+        final List<String> allowedCommands = Arrays.asList(ConfigCore.getParameter(
```

Comment:
Are cases considered when the configuration option is not set (`#activeMQ.kitodoScript.allow=...`) or not set to a value (`activeMQ.kitodoScript.allow=`)?

Reply:
Maybe you should use the `getStringArrayParameter()` method to retrieve the list as of strings. Then you did not even need to call the `split(",")` method as this is already done. I don't know how the handling of not set or empty set configuration value is.

Reply:
ConfigCore.getParameter() should NoSuchElementException. But if it would return `null`, String.split() would throw a NullPointerException, and if it would return empty string, any command would be forbidden. So it is safe.

Reply:
I don't like `getStringArrayParameter()` because it returns an array, not a list object, and—unintuitively—uses the ampersand as separator character.  I also couldn't figure out how it behaves with empty parameters. Do you really want that?

Reply:
I'm not a friend of reimplementing code again. Maybe move this code into a method of the `ConfigCore` class so the implementation can be reused on other places instead of hiding it in this place - maybe already the code is created on other places.

Reply:
I used the old code. We can then change this globally later, but the change would introduce differences in config, and I thought there was a reason not to use the separator comma for something.